File: Chapter7-2/Chapter7-2/ViewController.m
//
//  ViewController.m
//  Chapter7-2
//
//  Created by 南　京兵 on 2016/09/27.
//  Copyright © 2016年 南　京兵. All rights reserved.
//

#import "ViewController.h"
#import "Chapter7-Swift.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    MySwiftClass *msc = [[MySwiftClass alloc] init];
    msc.message = @"Swift";
    [msc messageRepeatBy: 10];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

@end

File: Chapter7-1/Chapter7-1/ViewController.swift
//
//  ViewController.swift
//  Chapter7-1
//
//  Created by 南　京兵 on 2016/09/27.
//  Copyright © 2016年 南　京兵. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        MyObjCModule().helloWorld() // SwiftからObjective-Cのメソッド呼び出し
        MyObjCModule().printString("This is ObjC method") // Swiftから引数のあるObjective-Cメソッド呼び出し
        MyObjCModule().printString(MySwiftModule.init(name: "John").name) //継承し初期化したnameをObjCのメソッドで表示
        let swiftModule = MySwiftModule.init(name: "shone") // 初期化
        swiftModule.helloWorld() // 継承したメソッドの呼び出し
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

class MySwiftModule : MyObjCModule {
    
    let name:String
    
    init(name: String) {
        self.name = name
    }
}

